src/Agent.Plugins/BuildDrop/BuildDropDropPlugin.cs

```diff
@@ -0,0 +1,234 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.BlobStore.Common;
+using Microsoft.VisualStudio.Services.Content.Common.Tracing;
+using Microsoft.VisualStudio.Services.BlobStore.WebApi;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.WebApi;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Agent.Sdk;
+
+namespace Agent.Plugins.BuildDrop
+{
+    public abstract class BuildDropTaskPluginBase : IAgentTaskPlugin
+    {
+        public abstract Guid Id { get; }
+        public abstract string Version { get; }
+        public abstract string Stage { get; }
+
+        public async Task RunAsync(AgentTaskPluginExecutionContext context, CancellationToken token)
+        {
+            ArgUtil.NotNull(context, nameof(context));
+
+            Guid projectId = new Guid(context.Variables.GetValueOrDefault(BuildVariables.TeamProjectId)?.Value ?? Guid.Empty.ToString());
+            ArgUtil.NotEmpty(projectId, nameof(projectId));
+
+            string buildIdStr = context.Variables.GetValueOrDefault(BuildVariables.BuildId)?.Value ?? string.Empty;
+            if (!int.TryParse(buildIdStr, out int buildId))
+            {
+                throw new ArgumentOutOfRangeException(buildIdStr);
+            }
+
+            string containerIdStr = context.Variables.GetValueOrDefault(BuildVariables.ContainerId)?.Value ?? string.Empty;
```

Comment:
why you need container id? we are not going to File Container at all right?

Reply:
removed